iOS/AppDelegate.swift

```diff
@@ -105,75 +105,54 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UISplitViewControllerDele
 				}
 			}
 		}
-
-		updateBackgroundRefreshInterval()
 		
 		syncTimer = ArticleStatusSyncTimer()
-
+		
 		#if DEBUG
-			syncTimer!.update()
+		syncTimer!.update()
 		#endif
 		
 		return true
 		
 	}
-
-//	func application(_ application: UIApplication, shouldSaveApplicationState coder: NSCoder) -> Bool {
-//
-//		let versionNumber = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
-//		coder.encode(versionNumber, forKey: "VersionNumber")
-//
-//		return true
-//
-//	}
-//
-//	func application(_ application: UIApplication, shouldRestoreApplicationState coder: NSCoder) -> Bool {
-//		if let storedVersionNumber = coder.decodeObject(forKey: "VersionNumber") as? String {
-//			let versionNumber = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
-//			if versionNumber == storedVersionNumber {
-//				return true
-//			}
-//		}
-//		return false
-//	}
+	
+	//	func application(_ application: UIApplication, shouldSaveApplicationState coder: NSCoder) -> Bool {
+	//
+	//		let versionNumber = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
+	//		coder.encode(versionNumber, forKey: "VersionNumber")
+	//
+	//		return true
+	//
+	//	}
+	//
+	//	func application(_ application: UIApplication, shouldRestoreApplicationState coder: NSCoder) -> Bool {
+	//		if let storedVersionNumber = coder.decodeObject(forKey: "VersionNumber") as? String {
+	//			let versionNumber = Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as! String
+	//			if versionNumber == storedVersionNumber {
+	//				return true
+	//			}
+	//		}
+	//		return false
+	//	}
 	
 	func applicationWillResignActive(_ application: UIApplication) {
 		// Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
 		// Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
 	}
-
+	
 	func applicationDidEnterBackground(_ application: UIApplication) {
 		syncTimer?.invalidate()
 		
-		let completeProcessing = { [unowned self] in
-			UIApplication.shared.endBackgroundTask(self.syncBackgroundUpdateTask)
-			self.syncBackgroundUpdateTask = UIBackgroundTaskIdentifier.invalid
-		}
-		
-		DispatchQueue.global(qos: .background).async { [unowned self] in
-			
-			self.syncBackgroundUpdateTask = UIApplication.shared.beginBackgroundTask {
-				completeProcessing()
-				os_log("Accounts sync processing terminated for running too long.", log: self.log, type: .info)
-			}
-			
-			DispatchQueue.main.async {
-				AccountManager.shared.syncArticleStatusAll() {
-					completeProcessing()
-				}
-				
-			}
-			
-		}
```

Comment:
What this code does is that it *immediately* does a sync of the article status.  It doesn't do a full refresh.  We just want it to quickly and immediately sync the article status.

Do we need to change anything in applicationDidEnterBackground?  I don't see that it is using anything deprecated.

Reply:
I've restored this code.